week3/src/components/CardContainer.jsx

```diff
@@ -0,0 +1,145 @@
+import { useMemo, useState, useEffect } from "react"
+import styled from 'styled-components';
+import CARDS from '../constants/card';
+import Card from "./Card";
+import ClearModal from "./ClearModal";
+import { LEVEL } from "../constants/Level";
+
+export default function CardContainer({updateScore}) {
+
+    // // 게임이 끝났는지 저장
+    const [finished, setFinished] = useState(false)
+    // 현재 선택한 카드 데이터 저장하는 배열
+    const [clicked, setClicked] = useState(Array.from([]))
+
+    // 게임이 끝나면 재배열 (finish 값이 변경되는 경우)
+    const MixedCardList = useMemo(() => MixedCardDeck(), [finished])
+
+    // 인덱스로 카드 선택
+    const handleClick = (idx) => {
+        // 선택한 카드 뒤집기
+        MixedCardList.some((e) => {
+            if (e.idx === idx) {
+            e.status = !e.status
+            return true
+        }
+          return false
+        })
+    
+        setClicked([...clicked, idx])
+      }
+    
+      useEffect(() => {
+        setTimeout(() => {
+          if (clicked.length === 2) {
+            let a = MixedCardList.find((e) => e.idx === clicked[0]).id
+            let b = MixedCardList.find((e) => e.idx === clicked[1]).id
+            
+            // 다른 경우 status를 false로 유지
+            if (a !== b) {
+              MixedCardList.forEach((e) => {
+                if (e.idx === clicked[0] || e.idx === clicked[1]) {
+                  e.status = false
+                }
+              })
+            } else{
+              // 같은 경우 점수 올리기
+              updateScore((currentScore) => currentScore + 1);
+            }
+            // 선택 끝나면 배열 비우기
+            setClicked(Array.from([]))
+          }
+        }, 700)
+      }, [clicked])
+    
+      useEffect(() => {
+        if (MixedCardList.every((e) => e.status === true)) {
+          setFinished(true)
+        }
+      }, [clicked])
+    
+      const handleFinished = () => {
+        setFinished(false)
+        updateScore(0);
+      }
+
+    return(
+        <GameLayout>
+            {finished && (
+            <ClearModal handleFinished={handleFinished} >
+            </ClearModal>
+      )}
+            <CardsLayout>
+                {MixedCardList.map((e) => (
+                <Card
+                    data={e}
+                    key={e.idx}
+                    handleClick={clicked.length < 2 ? handleClick : null}
```

Comment:
여기도 이렇게 작성해주는 게 더 좋은 패턴입니다!!
```suggestion
                    handleClick={clicked.length < 2 && handleClick}
```